components/TopNavigation.tsx

```diff
@@ -0,0 +1,24 @@
+import type { FC } from 'react';
+
+import ActiveLink from '@/components/Common/ActiveLink';
+import { useSiteNavigation } from '@/hooks/server';
+
+const TopNavigation: FC = () => {
+  const { navigationItems } = useSiteNavigation();
+
+  return (
+    <nav aria-label="primary">
+      <ul className="list-divider-pipe">
+        {navigationItems.map((item, key) => (
+          <li key={key}>
+            <ActiveLink href={item.link} allowSubPath>
+              {item.text}
+            </ActiveLink>
+          </li>
+        ))}
```

Comment:
```suggestion
        {navigationItems.map(({ link, text }) => (
          <li key={link}>
            <ActiveLink href={link} allowSubPath>
              {text}
            </ActiveLink>
          </li>
        ))}
```

IMO it is better to use the url instead of the index as a key.